gap/ClassicalMaximals.gi

```diff
@@ -1,10 +1,99 @@
 #
 # ClassicalMaximals: Maximal subgroups of classical groups
 #
+# Code along the lines of:
+# [1]   J. M. Bray, D. F. Holt, C. M. Roney-Dougal. "The Maximal Subgroups of the
+#       Low-Dimensional Finite Classical Groups". Cambridge UP, 2013.
+# [2]   D. F. Holt, C. M. Roney-Dougal. "Constructing Maximal Subgroups of
+#       Classical Groups." LMS Journal of Computation and Mathematics, vol. 8,
+#       2005, pp. 46-79.
+#
 # Implementations
 #
-InstallGlobalFunction( ClassicalMaximals_Example,
-function(x)
-	Print( "This is a placeholder function, replace it with your own code.\n" );
-end );
 
+ConjugatesInGeneralGroup := function(S, C, r)
+    return List([0..r - 1], i -> S ^ (C ^ i));
+end;
+
+InstallGlobalFunction(ClassicalMaximalsGeneric,
+function(type, n, q)
+    if type = "L" then
+        return MaximalSubgroupClassRepsSpecialLinearGroup(n, q);
+    fi;
+
+    return 0;
+end);
+
+InstallGlobalFunction(MaximalSubgroupClassRepsSpecialLinearGroup,
+function(n, q)
+    local maximalSubgroups, k, divisors, t, primeDivisorsOfn, primeDivisorsOfe, s, factorisation, p,
+    e, generatorGLMinusSL, degreeOfExtension, f, numberOfConjugates,
+    subfieldGroup;
+    maximalSubgroups := [];
+
+    if (n = 2 and q <= 3) then
+        Error("SL_2(2) and SL_2(3) are soluble");
+    fi;
+
+    factorisation := PrimePowersInt(q);
+    p := factorisation[1];
+    e := factorisation[2];
+    generatorGLMinusSL := GL(n, q).1;
+
+    # Class C1 subgroups
+    for k in [1..n-1] do
+        Add(maximalSubgroups, SLStabilizerOfSubspace(n, q, k));
```

Comment:
Is this always so? Or only in dimensions 2 and 3? 
For dimension 2 this is Prop 3.1.2. Let's add the relevant Theorem/Proposition/Lemma to each case.